src/trans.rs

```diff
@@ -549,6 +720,74 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
         }
     }
 
+    fn generate_runtime_start(&mut self, entry_fn: &str) -> BinaryenFunctionRef {
```

Comment:
yeah, it seems very platform specific as you say. I'm not yet clear on our libcore/libstd plan so we'll still have to talk about all this.